tests/foreman/rhci/test_computeresources.py

```diff
@@ -0,0 +1,353 @@
+         'provider': FOREMAN_PROVIDERS['vmware'],
+         'url': vmware_vcenter,
+         'username': vmware_username,
+         'password': vmware_password,
+         'datacenter': vmware_datacenter,
+         'url_name': vmware_url_name,
+         'dc_locator': vmware_dc_locator}
+    )
+    def test_retrieve_vm_list(self, data):
+        """ @Test: Retrieve list of VMs.
+        @Feature: Compute Resource
+        """
+        with Session(self.browser) as session:
+            make_resource(
+                session,
+                name=data['name'],
+                provider_type=data['provider'],
+                parameter_list=[
+                    [data['url_name'], data['url'], 'field'],
+                    ['Username', data['username'], 'field'],
+                    ['Password', data['password'], 'field'],
+                    [data['dc_locator'], data['datacenter'], 'special select']
+                ],
+                orgs=[self.default_org],
+                org_select=False,
+                locations=[self.default_loc],
+                loc_select=True
+            )
+            # nasty hack
+            time.sleep(5)
+            search = self.compute_resource.search(data['name'])
+            self.assertIsNotNone(search)
+
+            print "%s:" % data['name']
+            for item in self.compute_resource.list_vms(data['name']):
+                if item.text:
+                    print "VM: %s" % item.text
+
+            self.compute_resource.delete(data['name'])
+            search = self.compute_resource.search(data['name'])
+            self.assertIsNone(search)
+
+    @data(
+        {'name': vmware_name,
+         'provider': FOREMAN_PROVIDERS['vmware'],
+         'url': vmware_vcenter,
+         'username': vmware_username,
+         'password': vmware_password,
+         'datacenter': vmware_datacenter,
+         'url_name': vmware_url_name,
+         'dc_locator': vmware_dc_locator}
+    )
+    def test_retrieve_template_list(self, data):
+        """ @Test: Retrieve list of templates..
+        @Feature: Compute Resource
+        """
+        with Session(self.browser) as session:
+            make_resource(
+                session,
+                name=data['name'],
+                provider_type=data['provider'],
+                parameter_list=[
+                    [data['url_name'], data['url'], 'field'],
+                    ['Username', data['username'], 'field'],
+                    ['Password', data['password'], 'field'],
+                    [data['dc_locator'], data['datacenter'], 'special select']
+                ],
+                orgs=[self.default_org],
+                org_select=False,
+                locations=[self.default_loc],
+                loc_select=True
+            )
+            # nasty hack
+            time.sleep(5)
+            search = self.compute_resource.search(data['name'])
+            self.assertIsNotNone(search)
+
+            print "%s:" % data['name']
```

Comment:
Why do we need `print` statement here. It is unusual for us to have any output in the code
We use `self.assertxxxxxxx` for any kind of verifications